src/record_ts.cpp

```diff
@@ -2126,6 +2061,222 @@ void RecordThreadState::add_out_param(uint32_t slot, uint32_t vtype) {
+        }
+    }
+
+    void barrier() override { record_exception(); };
+    Task *launch(Kernel /*kernel*/, KernelKey * /*key*/, XXH128_hash_t /*hash*/,
+                 uint32_t /*size*/, std::vector<void *> * /*kernel_params*/,
+                 const std::vector<uint32_t> * /*kernel_param_ids*/) override {
+        record_exception();
+        return nullptr;
+    };
+    void memset_async(void * /*ptr*/, uint32_t /*size*/, uint32_t /*isize*/,
+                      const void * /*src*/) override {
+        record_exception();
+    };
+    uint32_t compress(const uint8_t * /*in*/, uint32_t /*size*/,
+                      uint32_t * /*out*/) override {
+        record_exception();
+        return 0;
+    };
+    uint32_t mkperm(const uint32_t * /*values*/, uint32_t /*size*/,
+                    uint32_t /*bucket_count*/, uint32_t * /*perm*/,
+                    uint32_t * /*offsets*/) override {
+        record_exception();
+        return 0;
+    };
+    void memcpy(void * /*dst*/, const void * /*src*/,
+                size_t /*size*/) override {
+        record_exception();
+    };
+    void memcpy_async(void * /*dst*/, const void * /*src*/,
+                      size_t /*size*/) override {
+        record_exception();
+    };
+    void block_reduce(VarType /*vt*/, ReduceOp /*op*/, uint32_t /*size*/,
+                      uint32_t /*block_size*/, const void * /*in*/,
+                      void * /*out*/) override {
+        record_exception();
+    };
+    void block_prefix_reduce(VarType /*vt*/, ReduceOp /*op*/, uint32_t /*size*/,
+                             uint32_t /*block_size*/, bool /*exclusive*/,
+                             bool /*reverse*/, const void * /*in*/,
+                             void * /*out*/) override {
+        record_exception();
+    };
+    void reduce_dot(VarType /*type*/, const void * /*ptr_1*/,
+                    const void * /*ptr_2*/, uint32_t /*size*/,
+                    void * /*out*/) override {
+        record_exception();
+    };
+    void poke(void * /*dst*/, const void * /*src*/,
+              uint32_t /*size*/) override {
+        record_exception();
+    };
+    void aggregate(void * /*dst*/, AggregationEntry * /*agg*/,
+                   uint32_t /*size*/) override {
+        record_exception();
+    };
+    void enqueue_host_func(void (* /*callback*/)(void *),
+                           void * /*payload*/) override {
+        record_exception();
+    };
+    void notify_expand(uint32_t /*index*/) override {};
+    void reduce_expanded(VarType /*vt*/, ReduceOp /*reduce_op*/,
+                         void * /*data*/, uint32_t /*exp*/,
+                         uint32_t /*size*/) override {};
+    void notify_free(const void * /*ptr*/) override {};
+};
+
+void disable_thread_state(ThreadState **ts, JitBackend recording_backend) {
+    if (!*ts)
+        return;
+    *ts = new DisabledThreadState(*ts, recording_backend);
+}
+
+void enable_thread_state(ThreadState **ts) {
```

Comment:
The naming of these functions is a bit weird because they exist to disable functionality instead of enabling it. Would a more logical names of these function be something like ``set_disabled_thread_state`` and ``unset_disabled_thread_state`?